databay/misc/buffers.py

```diff
@@ -8,10 +8,13 @@ class Buffer():
     def __init__(self,
                  count_threshold : int = None,
                  time_threshold: float = None,
-                 custom_controllers: Union[callable, List[callable]] = None):
+                 custom_controllers: Union[callable, List[callable]] = None,
+                 controller_conjunction: bool = False
+                 ):
         self.count_threshold = count_threshold
         self.time_threshold = time_threshold
         self.custom_controllers = custom_controllers
+        self.controller_conjunction = controller_conjunction
```

Comment:
Awesome that this feature is in there, it adds a nice little bit of extra fine grained control for the buffer.

Similar nit to the `Splitter` conversation 😅 but I think `controller_conjunction` is too verbose here and doesn't provide much clue as to what the actual action it invokes is. I think some sort of verb would be beneficial specifically because this is a `bool` flag so enabling it will either cause or prevent an action. 
```python
if self.controller_conjunction:
```
I feel this line demonstrates what I'm discussing, `controller_conjunction` as a bool flag doesn't immediately express whether `True` or `False` are natural stand ins for `AND` and `OR`. It feels like the attribute could be of any type, I think it should be a string when I read it. 

Some suggestions I have are below, granted not all of those work in the single line context either:

- `strict_release_enabled`
- `strict_release`
- `exclusive_release`
- `exclusive_release_enabled`
- `enable_strict_release` 
- `controller_release_mode`

Reply:
> I feel this line demonstrates what I'm discussing, controller_conjunction as a bool flag doesn't immediately express whether True or False

Cool, that's a handy way of looking at it, you're right!

As for your suggestions - I don't feel that either 'strict' or 'exclusive' are descriptive enough. Without looking at the documentation I wouldn't know what they do, as having a 'strict/exclusive controller/release' is not a widely accepted nomenclature and is up for intepretation.

I liked 'conjunction' as it is a very specific nomenclature from boolean algebra, literally being a synonym for `AND`. And so the opposite of conjunction (ie. not conjunction, or conjunction=False) is disjunction, which is also a specific boolean algebra term - a synonym for `OR`. More here on [Wikipedia on Boolean Algebra](https://en.wikipedia.org/wiki/Boolean_algebra#Basic_operations)

We could turn it into a verb like you suggest and say `self.conjugate_controllers`?

Feel free to suggest other alternatives though - if you can think of something better than `conjugate_controllers` then I'd gladly take that as it does sounds a bit ugly 😅

Reply:
Yeah this is another hard one to find the right terminology for. I'm familiar with conjunctions and disjunctions but I just don't see the terminology used often in library's that are doing boolean operations, while the terms are appropriate for me they still seem a bit overly wordy and academic to use in the API implementation. That being said I am struggling to think of an alternative.  I know Numpy has a bunch of [logical functions](https://numpy.org/doc/stable/reference/routines.logic.html) maybe we could possibly take some inspiration from one of those implementations. If not feels appropriate from that though I say let's go with `conjugate_controllers` anyway as it describes what's happening.

Reply:
You're right - not frequently used in libraries, and indeed quite academic. I'm struggling for an alternative too man, damn these things are all so hard to name! 😄 I've had a look at the numpy logical functions you linked, but I don't think any of these would suit our case. `conjugate_controllers` it is then 👍 Thanks for bringing the discussion on the table though, as it is definitely better than `controller_conjunction` 👏